directlineclient.js

```diff
@@ -28,14 +30,34 @@ DirectLineClient.prototype.init = function(context, testData) {
         headers: headers,
         json: true
     };
+
     logger.log(`Init conversation request: ${JSON.stringify(startConversationOptions)}`);
-    var promise = rp(startConversationOptions)
-        .then(function(response) {
-            logger.log("init response: " + utils.stringify(response));
-            self.watermark[response.conversationId] = null;
-            self.headers[response.conversationId] = headers;
-            return response;
-        });
+    // retrying init:
+    let result = false;
+    let message;
+    let promise;
+    let tolerance = parseInt(process.env.failureTolerance) ? parseInt(process.env.failureTolerance) : config.defaults.failureTolerance;
+    let iterNum = tolerance;
+    while (!result && (tolerance > 0)) {
+        promise = await rp(startConversationOptions)
+            .then(function(response) {
+                message = utils.stringify(response)
+                logger.log("init response: " + message);
+                self.watermark[response.conversationId] = null;
+                self.headers[response.conversationId] = headers;
+                result = true;

```

Comment:
Consider changing 'result' to 'success' and 'promise' to 'result. It would make more sense.